lib/galaxy/tool_util/deps/requirements.py

```diff
@@ -204,20 +217,97 @@ def __str__(self):
         return f"ContainerDescription[identifier={self.identifier},type={self.type}]"
 
 
+ResourceType = Literal[
+    "cores_min",
+    "cores_max",
+    "ram_min",
+    "ram_max",
+    "tmpdir_min",
+    "tmpdir_max",
+]
+VALID_RESOURCE_TYPES = get_args(ResourceType)
+
+
+class ResourceRequirement:
+    def __init__(self, value_or_expression: Union[int, float, str], resource_type: ResourceType):
+        self.value_or_expression = value_or_expression
+        if not resource_type:
+            raise ValueError("Missing resource requirement type")
+        if resource_type not in VALID_RESOURCE_TYPES:
+            raise ValueError(f"Invalid resource requirement type '{resource_type}'")
+        self.resource_type = resource_type
+        self._runtime_required: Optional[bool] = None
+
+    @property
+    def runtime_required(self):
+        if self._runtime_required is None:
+            try:
+                float(self.value_or_expression)
+                self._runtime_required = False
+            except ValueError:
+                self._runtime_required = True
+        return self._runtime_required
```

Comment:
Can we simplify here?

```suggestion
        try:
            float(self.value_or_expression)
            self.runtime_required = False
        except ValueError:
            self.runtime_required = True
```